File: core/src/test/java/com/jlpka/langidentify/AccentRemoverTest.java
package com.jlpka.langidentify;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class AccentRemoverTest {

  private final AccentRemover remover = new AccentRemover();

  @Test
  void asciiPassesThrough() {
    assertEquals("hello world", remover.remove("hello world").toString());
  }

  @Test
  void emptyString() {
    assertEquals("", remover.remove("").toString());
  }

  @Test
  void basicAccents() {
    assertEquals("cafe", remover.remove("café").toString());
    assertEquals("naive", remover.remove("naïve").toString());
    assertEquals("uber", remover.remove("über").toString());
  }

  @Test
  void ligatures() {
    assertEquals("ae", remover.remove("æ").toString());
    assertEquals("oe", remover.remove("œ").toString());
    assertEquals("ss", remover.remove("ß").toString());
  }

  @Test
  void typographicLigatures() {
    assertEquals("ff", remover.remove("ﬀ").toString());
    assertEquals("fi", remover.remove("ﬁ").toString());
    assertEquals("fl", remover.remove("ﬂ").toString());
    assertEquals("ffi", remover.remove("ﬃ").toString());
  }

  @Test
  void mixedContent() {
    assertEquals("Francais", remover.remove("Français").toString());
    assertEquals("Strasse", remover.remove("Straße").toString());
    assertEquals("resume", remover.remove("résumé").toString());
  }

  @Test
  void unmappedNonAsciiPassesThrough() {
    // CJK characters should pass through unchanged
    String cjk = "漢字";
    assertEquals(cjk, remover.remove(cjk).toString());
  }

  @Test
  void returnsInputForAllAscii() {
    // Fast path: all-ASCII input should return the original CharSequence
    String ascii = "plain ascii text";
    assertSame(ascii, remover.remove(ascii));
  }
}

File: rust/langidentify/langidentify/tests/accent_remover_test.rs
use langidentify::accent_remover::AccentRemover;

fn remover() -> AccentRemover {
    AccentRemover::new()
}

#[test]
fn ascii_passes_through() {
    assert_eq!("hello world", remover().remove("hello world").as_ref());
}

#[test]
fn empty_string() {
    assert_eq!("", remover().remove("").as_ref());
}

#[test]
fn basic_accents() {
    let r = remover();
    assert_eq!("cafe", r.remove("café").as_ref());
    assert_eq!("naive", r.remove("naïve").as_ref());
    assert_eq!("uber", r.remove("über").as_ref());
}

#[test]
fn ligatures() {
    let r = remover();
    assert_eq!("ae", r.remove("æ").as_ref());
    assert_eq!("oe", r.remove("œ").as_ref());
    assert_eq!("ss", r.remove("ß").as_ref());
}

#[test]
fn typographic_ligatures() {
    let r = remover();
    assert_eq!("ff", r.remove("\u{FB00}").as_ref());
    assert_eq!("fi", r.remove("\u{FB01}").as_ref());
    assert_eq!("fl", r.remove("\u{FB02}").as_ref());
    assert_eq!("ffi", r.remove("\u{FB03}").as_ref());
}

#[test]
fn mixed_content() {
    let r = remover();
    assert_eq!("Francais", r.remove("Français").as_ref());
    assert_eq!("Strasse", r.remove("Straße").as_ref());
    assert_eq!("resume", r.remove("résumé").as_ref());
}

#[test]
fn unmapped_non_ascii_passes_through() {
    let r = remover();
    let cjk = "漢字";
    assert_eq!(cjk, r.remove(cjk).as_ref());
}

#[test]
fn returns_borrowed_for_all_ascii() {
    let r = remover();
    let ascii = "plain ascii text";
    let result = r.remove(ascii);
    // Cow::Borrowed means no allocation occurred
    assert!(matches!(result, std::borrow::Cow::Borrowed(_)));
}
